examples/2gcheckerboard/base_2gcheckerboard.py

```diff
@@ -0,0 +1,196 @@
+    sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../../")))
+    from pyopensn.mesh import OrthogonalMeshGenerator
+    from pyopensn.xs import MultiGroupXS
+    from pyopensn.source import VolumetricSource
+    from pyopensn.aquad import GLCProductQuadrature2DXY
+    from pyopensn.solver import DiscreteOrdinatesProblem, SteadyStateSolver
+    from pyopensn.fieldfunc import FieldFunctionInterpolationVolume
+    from pyopensn.logvol import RPPLogicalVolume
+
+if __name__ == "__main__":
+
+    try:
+        print("Absorber Sigma_a Parameter = {}".format(abs_1))
+        param = abs_1
+    except:
+        abs_1=10.0
+        print("Absorber Sigma_a Nominal = {}".format(abs_1))
+
+    try:
+        print("Absorber Sigma_s Parameter = {}".format(scatt_1))
+        param = scatt_1
+    except:
+        scatt_1=10.0
+        print("Absorber Sigma_s Nominal = {}".format(scatt_1))
+
+
+    try:
+        print("Scatterer Sigma_a Parameter = {}".format(abs_2))
+        param = abs_2
+    except:
+        abs_2=0.0
+        print("Scatterer Sigma_a Nominal = {}".format(abs_2))
+
+    try:
+        print("Scatterer Sigma_s Parameter = {}".format(scatt_2))
+        param = scatt_2
+    except:
+        scatt_2=1.0
+        print("Scatterer Sigma_s Nominal = {}".format(scatt_2))
+
+
+    try:
+        print("Source Parameter = {}".format(param_q))
+        param = param_q
+    except:
+        param_q=1.0
+        print("Source Nominal = {}".format(param_q))
+
+    try:
+        print("Parameter id = {}".format(p_id))
+    except:
+        p_id=0
+        print("Parameter id = {}".format(p_id))
+
+    try:
+        if phase == 0:
+            print("Offline Phase")
+            phase = "offline"
+        elif phase == 1:
+            print("Merge Phase")
+            phase = "merge"
+        elif phase == 2:
+            print("Systems Phase")
+            phase = "systems"
+        elif phase == 3:
+            print("Online Phase")
+            phase = "online"
+    except:
+        phase="offline"
+        print("Phase default to offline")
+
+    # Check number of processors
+    num_procs = 4
+    if size != num_procs:
+        sys.exit(f"Incorrect number of processors. Expected {num_procs} processors but got {size}.")
+
+    # Setup mesh
+    nodes = []
+    N = 70
+    L = 7
+    xmin = 0
+    dx = L / N
+    for i in range(N + 1):
+        nodes.append(xmin + i * dx)
+    meshgen = OrthogonalMeshGenerator(node_sets=[nodes, nodes],
+        partitioner=KBAGraphPartitioner(
+            nx=2,
+            ny=2,
+            xcuts=[3.5],
+            ycuts=[3.5],
+        ))
+    grid = meshgen.Execute()
+
+    # Set background (Scatterer) block ID = 0
+    vol0 = RPPLogicalVolume(infx=True, infy=True, infz=True)
+    grid.SetBlockIDFromLogicalVolume(vol0, 0, True)
+
+    # Set Source (central red square from x=3 to x=4, y=3 to y=4) block ID = 1
+    vol_source = RPPLogicalVolume(xmin=3.0, xmax=4.0,
```

Comment:
could you rename vol_source to logvol_src or something like that, so that people do not think that python var holds the values of the vol_source